src/server/lib/members-data-api/user-attributes.ts

```diff
@@ -79,12 +79,7 @@ export const getUserAttributes = async (
       );
     }
 
-    const unparsed = await response.json();
-
-    const userAttributesResponse =
-      await userAttributesResponseSchema.parseAsync(unparsed);
-
-    return userAttributesResponse;
+    return userAttributesResponseSchema.parse(await response.json());
```

Comment:
I wonder what we could do if the json we receive doesn't match the schema exactly and `parse()` throws but it still has usable data 🤔 .. do we consider that to be an error, or can we still extract the critical parts that allow us to proceed?

Reply:
Zod only looks out for the fields that you've defined in the schema, so as long as we get back the fields that are defined, zod will only attempt to parse those.

It's unlikely that Okta will change their API responses, as it's versioned, and we've marked any possible missing field as nullable or optional, so we're pretty safe.

Reply:
aha , great point - if fields are removed on Oktas’ end we should probably error anyway as it could be a very bad thing 👍